### PR TITLE
Feature/39 add pvr functions to the application

### PR DESCRIPTION
**Pull request by apattnaik0721013**
1. Play, pause and stop buttons are added to the viewer.
2. Currently the functions of stop and pause buttons are identical.

### CODE REVIEW

File: include/System.h [hunk 1]
```diff
@@ -47,6 +47,12 @@ class Tracking;
 class LocalMapping;
 class LoopClosing;
 
+enum RUN_STATUS{
```
Comment (paul-michalik) [inline include/System.h]: - Use `enum class`
- Rename to `AppState`.  We never use upper case for anything except preprocessor controlled content. Actually I'd strongly recommend to use the C++ native naming convention, i.e. `enum class app_state` etc.
```

enum class AppState {
    Playing,
    Paused,
    Stopped
};
```

File: include/System.h [hunk 2]
```diff
@@ -47,6 +47,12 @@ class Tracking;
...
+    PLAYING=0,
```
Comment (paul-michalik) [inline include/System.h]: `Playing`, `Paused`, `Stopped`...

File: include/System.h [hunk 3]
```diff
@@ -125,6 +131,8 @@ class System
 	
 	KeySemanticObjGrp* GetSemanticObjGrp();
 	void SetSemanticObjGrpContent(traffic_sign_map_t const & InterestedObject);
+    void SetCurrentRunStatus(RUN_STATUS CurrentRunStatus);
```
Comment (paul-michalik) [inline include/System.h]: Use the event semantics for methods:
```
void Play();
void Resume()
void Stop();
```
Each method logically corresponds to an event sent to the state machine.

File: include/System.h [hunk 4]
```diff
@@ -178,6 +186,7 @@ class System
     std::vector<cv::KeyPoint> mTrackedKeyPointsUn;
     std::mutex mMutexState;
 	KeySemanticObjGrp mSemanticObjGrp;
+    RUN_STATUS mCurrentRunStatus;
```
Comment (paul-michalik) [inline include/System.h]: Modify the names accordingly.

File: src/System.cc [hunk 5]
```diff
@@ -52,6 +52,7 @@ System::System(const string &strVocFile, const string &strSettingsFile, const eS
     "under certain conditions. See LICENSE.txt." << endl << endl;
 
     cout << "Input sensor was set to: ";
+    mCurrentRunStatus = RUN_STATUS::PLAYING;
```
Comment (paul-michalik) [inline src/System.cc]: According to the state machine diagram the application should start in the state "Stopped"! Also set the state in the constructor initializer list.

File: src/Viewer.cc [hunk 6]
```diff
@@ -152,6 +155,22 @@ void Viewer::Run()
             mpSystem->Reset();
             menuReset = false;
         }
+        else if(menuPlay)
+        {
+            menuPlay = false;
+            mpSystem->SetCurrentRunStatus(RUN_STATUS::PLAYING);
```
Comment (paul-michalik) [inline src/Viewer.cc]: `mpSystem->Play();`

File: src/Viewer.cc [hunk 7]
```diff
@@ -152,6 +155,22 @@ void Viewer::Run()
...
+        }
+        else if(menuPause)
+        {
+            menuPause = false;
+            mpSystem->SetCurrentRunStatus(RUN_STATUS::PAUSED);
```
Comment (paul-michalik) [inline src/Viewer.cc]: `mpSystem->Pause();`

File: src/Viewer.cc [hunk 8]
```diff
@@ -152,6 +155,22 @@ void Viewer::Run()
...
+        }
+        else if(menuStop)
+        {
+            menuStop = false;
+            mpSystem->SetCurrentRunStatus(RUN_STATUS::PAUSED);
```
Comment (paul-michalik) [inline src/Viewer.cc]: `mpSystem->Pause();`

File: src/System.cc [hunk 9]
```diff
@@ -528,5 +530,14 @@ void System::SetSemanticObjGrpContent(traffic_sign_map_t const &InterestedObject
 {
 	mSemanticObjGrp.SetSemanticObjGrpContent(InterestedObject);
 }
+void System::SetCurrentRunStatus(RUN_STATUS CurrentRunStatus)
```
Comment (paul-michalik) [inline src/System.cc]: Modify according to state machine semantics, e.g.:
```
void System::Play()
{ 
    // do whatever is required to switch state according to state machine model
}
```

File: src/System.cc [hunk 10]
```diff
@@ -528,5 +530,14 @@ void System::SetSemanticObjGrpContent(traffic_sign_map_t const &InterestedObject
...
+{
+    mCurrentRunStatus = CurrentRunStatus;
+}
+void System::RunEventLoop()
```
Comment (paul-michalik) [inline src/System.cc]: This implementation is a little odd. I'd prefer a cleaner way - e.g. via a [condition variable](http://en.cppreference.com/w/cpp/thread/condition_variable). If you want to keep this implementation then at least rename this to something like `SpinUntilSignalled`...

**Comment by paul-michalik** [comment]
Please be a little more careful about the language and text formatting. Correct the description of the PR!

**Comment by apattnaik0721013** [comment]
@paul-michalik 
1. Address all your comments.
2. At the start , it will be in stopped state.
3. user need to press , play button to start the tracking.

File: src/System.cc (hunk 10 above)
Comment (paul-michalik) [reply src/System.cc]: I am renaming this to `WaitIfBlocked`

**Pull request merged**